### PR TITLE
Add Config.BossGrades

### PR DESCRIPTION
**Pull request by Lunar-Scripts**
Added the option to allow other grades than 'boss' to access the boss menu.
Could for example be used to allow an player with an 'underboss' grade to access the boss menu.

### CODE REVIEW

**Review by Taavi-AU — APPROVED** [review]

**Comment by wobozkyng** [comment]
this will need more adjustment for other resources, e.g. esx_policejob has client control for job grades to access boss menu.

File: [esx_addons]/esx_society/config.lua [hunk 1]
```diff
@@ -3,3 +3,6 @@ Config = {}
 Config.Locale = GetConvar('esx:locale', 'en')
 Config.EnableESXIdentity = true
 Config.MaxSalary = 3500
+Config.BossGrades = {
+    'boss'
+}
```
Comment (CsokiHUN) [inline [esx_addons]/esx_society/config.lua:8]:
```suggestion
Config.BossGrades = {
  ['boss'] = true
}
```

File: [esx_addons]/esx_society/server/main.lua [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
 local RegisteredSocieties = {}
 
+function IsGradeBoss(grade)
+	for k,v in ipairs(Config.BossGrades) do
+		if grade == v then
+			return true
+		end
+	end
+	return false
+end
```
Comment (CsokiHUN) [inline [esx_addons]/esx_society/server/main.lua:11]:
```suggestion
function IsGradeBoss(grade)
  return Config.BossGrades[grade]
end
```

**Review by CsokiHUN — CHANGES_REQUESTED** [review]

**Comment by CsokiHUN** [comment]
unnecessary for loop, please change to my solution

**Review by xorzo1 — APPROVED** [review]

**Comment by Benzo00** [comment]
Closed due to changes not being added, also will break other resources. please reach out when changes are made and i will reopen this

**Pull request closed without merging**